webapp/components/post_view/post_list.jsx

```diff
@@ -437,7 +441,12 @@ export default class PostList extends React.PureComponent {
         const posts = this.props.posts;
         const channel = this.props.channel;
 
-        if (posts == null || channel == null) {
+        if (posts == null) {
+            if (channel == null) {
+                const townsquare = ChannelStore.getByName('town-square');
+                browserHistory.push(TeamStore.getCurrentTeamRelativeUrl() + '/channels/' + townsquare.name);
```

Comment:
You can't perform any actions in a render() method, this needs to be handled elsewhere. Probably in the action that causes the channel to become `null`